sources/models/src/lib.rs

```diff
@@ -270,6 +270,8 @@ struct HostContainer {
 #[model]
 struct NetworkSettings {
     hostname: ValidLinuxHostname,
+    // Ordering matters in /etc/hosts if multiple domains point to the same IP, so we use BTreeMap to preserve order.
+    hosts: BTreeMap<IpAddr, Vec<ValidLinuxHostname>>,
     https_proxy: Url,
```

Comment:
I've been doing some extra reading about `BTreeMap` and my assumption that it preserves iteration order based on insertion is false: BTreeMap preserves iteration order based on key order. The common solution in the Rust community is to use [IndexMap](https://crates.io/crates/indexmap) for a map which preserves insertion order.


This also means that I would need to assert that every serializer and deserializer which touches settings objects respects ordering. Fortunately for `serde_json` and `toml` this can be enabled by adding a feature flag, but we have a few points e.g. in `migrations` where helpers assume that a `Map` can be safely represented via a HashMap.


I'm looking into solving this either by building in order preservation to settings (which could be extremely difficult to continuously enforce in the Bottlerocket repo, even with great testing in the core settings path) or by changing our model here to use an ordered data structure, although this adds some uncomfortable ergonomic questions over how to use the feature.